diff --git a/setup.py b/setup.py
index db77aa82..238ffd1f 100755
--- a/setup.py
+++ b/setup.py
@@ -54,7 +54,7 @@ class cmake_build(build):
         """
         Config and build third party dependencies.
         """
-        build_dir = os.path.join(root_dir, "third_party/build");
+        build_dir = "third_party/build"
         if not os.path.isdir(build_dir):
             os.makedirs(build_dir);

@@ -65,12 +65,13 @@ class cmake_build(build):
         check_call(command.split());
         command = "cmake --build . --target install";
         check_call(command.split());
+        os.chdir('../..')

     def build_pymesh(self):
         """
         Config and build pymesh.
         """
-        build_dir = os.path.join(root_dir, "build");
+        build_dir = "build"
         if not os.path.isdir(build_dir):
             os.makedirs(build_dir);

@@ -78,9 +79,9 @@ class cmake_build(build):
         command = "cmake .. -DCMAKE_BUILD_TYPE=Release -DPythonLibsNew_FIND_VERSION={v[0]}.{v[1]}".format(
                 v=platform.python_version_tuple());
         check_call(command.split());
-        command = "cmake --build . --config Release";
+        command = "cmake --build . --config Release -- -j {}".format(num_cores);
         check_call(command.split());
-        os.chdir(root_dir);
+        os.chdir('..');

     def run(self):
         self.build_third_party();
